### ISSUE TITLE
/home/friends sort option highlighting doesn't appear when switching sorts

### ISSUE THREAD

**Issue by tybug** [ppy/osu-web#7563]
The default sort option has a purple highlight:
![image](https://user-images.githubusercontent.com/31628143/117373678-b54ec600-ae99-11eb-9c72-bc75c8471fc0.png)

when switching to another one, it bolds, but doesn't highlight: 
![image](https://user-images.githubusercontent.com/31628143/117373728-d1eafe00-ae99-11eb-9f44-9029f7d7cd9a.png)

it highlights properly once I click anywhere on the page.

**Comment by tybug** [comment]
Interestingly this also happens in multiple places on https://osu.ppy.sh/beatmapsets - 

on any of the search options, clicking one doesn't turn it white until you click again anywhere on the page 
![image](https://user-images.githubusercontent.com/31628143/117374153-a1f02a80-ae9a-11eb-8237-8a447cda39a9.png) (this is the page after I selected osu!catch)

and again for sorting ("difficulty" should be blue) 
![image](https://user-images.githubusercontent.com/31628143/117374261-d4018c80-ae9a-11eb-9448-fa2df24db654.png)

**Comment by nanaya** [comment]
Probably more like because `:focus` is styled the same as `:hover` and `:active` (almost?) everywhere. Ideally those should all be different but then @arflyte will need to provide all four states (or five if including disabled).

**Comment by kj415j45** [comment]
Just notice this bug, and do some test around it. All references are from MDN.
> The `:focus` CSS pseudo-class represents an element (such as a form input) that has received focus. It is generally triggered when the user clicks or taps on an element or selects it with the keyboard's `Tab` key.

As far as I known, click/tap on element will make it `:focus` immediately, and it would only de-`:focus` when you click/tap on other element.

---

> The `:hover` CSS pseudo-class matches when the user interacts with an element with a pointing device, but does not necessarily activate it. It is generally triggered when the user hovers over an element with the cursor (mouse pointer).

For desktop, as above.
For mobile, element is `:active` when you long press the element but move your cursor later, see `:active` below. But remember:
> The `:hover` pseudo-class is problematic on touchscreens. Depending on the browser, the `:hover` pseudo-class might never match, match only for a moment after touching an element, or continue to match even after the user has stopped touching and until the user touches another element. Web developers should make sure that content is accessible on devices with limited or non-existent hovering capabilities.

---

> The `:active` CSS pseudo-class represents an element (such as a button) that is being activated by the user. When using a mouse, "activation" typically starts when the user presses down the primary mouse button.

For desktop, element is `:active` between your mouse click and release, and drag it to anywhere doesn't matter.
For mobile, element is `:active` when you long press the element without moving your cursor (or say, finger). And the element may not de-`:active` or de-`:hover` until you touch any otherwhere.

---

> Styles defined by the `:active` pseudo-class will be overridden by any subsequent link-related pseudo-class (`:link`, `:visited`, or `:active`) that has at least equal specificity. To style links appropriately, put the `:hover` rule after the `:link` and `:visited` rules but before the :active one, as defined by the LVHA-order: `:link` — `:visited` — `:hover` — `:active`.

---

I used following CSS to test my expound, on Chrome 90.0.4430.212 for Windows (10):
```css
.sort__item--active:focus {
    color: red;
}

.sort__item--active:hover {
    color: blue;
}

.sort__item--active:active {
    color: orange;
}
```

I'm not sure if `:focus` should be the first one, but put it at the bottom make it strange.

**Comment by Crystal-RainSlide** [comment]
The behavior of `:focus` is inconsist between browsers:
- https://github.com/GoogleChrome/web.dev/issues/1047
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#clicking_and_focus

I think the most influential one is, Safari won't give focus to the `<a>`s and `<button>`s without a `tabindex` attribute, when clicking them with a pointer device. It's just like the [Scrollbar Blindness](https://svenkadak.com/blog/scrollbar-blindness).

<details><summary>Other words</summary>

Focus is more of a indicator of current element when navgating though keyboard, instead of a "just clicked" indicator. Also see [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/%3Afocus-visible). Sadly, navgating the OSU website with keyboard is not very possible without, a, screen reader.

There is also [`:focus-within`](https://developer.mozilla.org/en-US/docs/Web/CSS/%3Afocus-within) for styling focus of child elements.

Both `:focus-visible` and `:focus-within` are new.

</details>

**Comment by nanaya** [comment]
This is more or less fixed by #8590.
